trussworks/react-uswds
# build: Update yarn publish scripts, move dependencies to devDependencies

# Summary

This PR makes some minor changes to our [npm lifecycle scripts](https://docs.npmjs.com/cli/v6/using-npm/scripts#life-cycle-scripts):
- Use `prepare` to build the library -- this will build the `/lib` directory before publishing, and on local `yarn install`, and before installing from a git tag, resolving https://github.com/trussworks/react-uswds/issues/312
- Adds `prepublishOnly` to ensure tests and linters are passing before publishing

It also moves dependencies into `devDependencies` and adds `uswds` to `peerDependencies`.

I also updated the README because I discovered an issue if you try to use JS components (such as the ComboBox or DatePicker) but are also importing the full USWDS JS in your project (`import 'uswds'`). This will result in the UI initializing twice and it's very buggy.

## Related Issues or PRs

Closes https://github.com/trussworks/react-uswds/issues/312

## How To Test

This PR should be tested by installing this branch on applications and making sure use of ReactUSWDS components still works as it did before. To do this, edit your `package.json` file and point the library to this branch:

```
"@trussworks/react-uswds": "https://github.com/trussworks/react-uswds#sr-build-updates"
```

Then run `yarn install` and start your app as usual. The install might take a little longer than usual, because it has to compile the library since it's not using a published version. You can also verify the correct version is installed by looking in your `yarn.lock` file. The `@trussworks/react-uswds` entry should look like this, with "resolved" pointing to this repo instead of the Yarn package registry:

```
"@trussworks/react-uswds@https://github.com/trussworks/react-uswds#sr-build-updates":
  version "1.10.0"
  resolved "https://github.com/trussworks/react-uswds#3626600feb13f9e5e58a86e6bfa72e450f8123fe"
```
